src/azure-cli/azure/cli/command_modules/redis/custom.py

```diff
@@ -72,6 +74,11 @@ def cli_redis_update(cmd, instance, sku=None, vm_size=None):
     return update_params
 
 
+def custom_update_setter(client, resource_group_name, name, parameters):
+    # Custom update setter is used to match behavior from when update was not a LRO
+    return client.begin_update(resource_group_name, name, parameters).result(0)
```

Comment:
According to the definition of `azure.core.polling._poller.LROPoller.result`:

https://github.com/Azure/azure-sdk-for-python/blob/d678ef6e5ebe71bcc7bb7f3665d76576437f0121/sdk/core/azure-core/azure/core/polling/_poller.py#L242-L252

```py
    def result(self, timeout: Optional[float] = None) -> PollingReturnType_co:
        """Return the result of the long running operation, or
        the result available after the specified timeout.


        :param float timeout: Period of time to wait before getting back control.
        :returns: The deserialized resource of the long running operation, if one is available.
        :rtype: any or None
        :raises ~azure.core.exceptions.HttpResponseError: Server problem with the query.
        """
        self.wait(timeout)
        return self._polling_method.resource()
```

The `0` in `result(0)` makes the `LROPoller` return immediately, causing CI failure:

https://dev.azure.com/azclitools/public/_build/results?buildId=121462&view=logs&j=038813f6-b85b-5e37-acf4-30b28352865f&t=48a22e66-ceab-5c7f-d00b-cb2046709dfd

```
2024-01-12T08:59:22.0016933Z =================================== FAILURES ===================================
2024-01-12T08:59:22.0017289Z ___________________ RedisCacheTests.test_redis_cache_update ____________________
2024-01-12T08:59:22.0019920Z [gw2] linux -- Python 3.11.5 /opt/az/bin/python3
2024-01-12T08:59:22.0020425Z /opt/az/lib/python3.11/site-packages/azure/cli/testsdk/base.py:145: in tearDown
2024-01-12T08:59:22.0020723Z     super(ScenarioTest, self).tearDown()
2024-01-12T08:59:22.0020984Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2024-01-12T08:59:22.0021110Z 
2024-01-12T08:59:22.0021390Z self = <command_modules.redis.tests.latest.test_redis_scenario.RedisCacheTests testMethod=test_redis_cache_update>
2024-01-12T08:59:22.0021580Z 
2024-01-12T08:59:22.0021788Z     def tearDown(self):
2024-01-12T08:59:22.0022010Z         os.environ = self.original_env
2024-01-12T08:59:22.0022239Z         # Autorest.Python 2.x
2024-01-12T08:59:22.0022527Z         assert not [t for t in threading.enumerate() if t.name.startswith("AzureOperationPoller")], \
2024-01-12T08:59:22.0022994Z             "You need to call 'result' or 'wait' on all AzureOperationPoller you have created"
2024-01-12T08:59:22.0023274Z         # Autorest.Python 3.x
2024-01-12T08:59:22.0027370Z >       assert not [t for t in threading.enumerate() if t.name.startswith("LROPoller")], \
2024-01-12T08:59:22.0028218Z             "You need to call 'result' or 'wait' on all LROPoller you have created"
2024-01-12T08:59:22.0028728Z E       AssertionError: You need to call 'result' or 'wait' on all LROPoller you have created
2024-01-12T08:59:22.0028890Z 
2024-01-12T08:59:22.0029292Z /opt/az/lib/python3.11/site-packages/azure/cli/testsdk/scenario_tests/base.py:160: AssertionError
2024-01-12T08:59:22.0029744Z ------------- generated xml file: /azure_cli_test_result/redis.xml -------------
2024-01-12T08:59:22.0030074Z =========================== short test summary info ============================
2024-01-12T08:59:22.0030422Z FAILED tests/latest/test_redis_scenario.py::RedisCacheTests::test_redis_cache_update
```

No timeout should be given. `result()` should be used instead, such as:

https://github.com/Azure/azure-cli/blob/573410a8a39597f0181e942eb011bd131fee0d72/src/azure-cli/azure/cli/command_modules/redis/custom.py#L231

Reply:
I think it's be design to make the operation returning immediately. But they should use `sdk_no_wait` instead of `result(0)`

Reply:
`sdk_no_wait` turns off polling:

https://github.com/Azure/azure-cli/blob/a5198b578b17de934e15b1c92e369e45323e9658/src/azure-cli-core/azure/cli/core/util.py#L707-L710

but `result(0)` doesn't. `result(0)` will make the result returned immediately, but SDK still keeps polling in the background.

Reply:
Hi @jiasli and @evelyn-ys, thanks for your work identifying this issue. Is the expectation that the service team make this change or is this going to be handled by the CLI team?

Reply:
appreciated that @austintolani you can create another PR to fix!

Reply:
Okay thanks @evelyn-ys.  I am no longer on the Redis team so I don't think it's appropriate for me to make this change but I have forwarded this information to the Redis sub-team that owns the CLI.